### DISCUSSION TITLE
macOS Sonoma, Yubikey doesn't work with ssh-keygen

### DISCUSSION THREAD

**Discussion by miklosakos** [Yubico/libfido2#753] [Q&A]
**What version of libfido2 are you using?**
1.14.0
**What operating system are you running?**
macOS Sonoma 14.1.1 [arm64] 
**What application are you using in conjunction with libfido2?**
macOS ssh client
**How does the problem manifest itself?**
Generating public/private ed25519-sk key pair.
You may need to touch your authenticator to authorize key generation.
No FIDO SecurityKeyProvider specified
Key enrollment failed: invalid format
**Is the problem reproducible?**

**What are the steps that lead to the problem?**
wanting to create a FIDO SSH key

<details>
<summary><code>fido2-token -L</code></summary>
<br>
<pre>
$ fido2-token -L
ioreg://4295492886: vendor=0x1050, product=0x0407 (Yubico YubiKey OTP+FIDO+CCID)
</pre>
</details>

**Comment by LDVG** [comment]
Hi,

This thread was converted from an issue since this is not a bug with libfido2.

The  version of OpenSSH that is bundled with macOS does not include the built-in security key provider (aka middleware). You can either build a standalone version of the provider yourself, or use OpenSSH from brew. You may find more details in the discussion following https://github.com/Yubico/libfido2/issues/464#issuecomment-1577748167.

Let us know if you've got any more questions.
